lib/ai.rb

```diff
@@ -3,66 +3,81 @@
+  end
+ 
+  def possible_rows(ship, orientation)
+    orientation == "horizontal" ? rows_data[0..9] : rows_data[0..(10-ship.length)]
+  end
 
-      # battleship
-    fleet_placement_board.ships[1].segments[0].set_coordinate(Coordinate.new("b","1")).mark_as_hit
-    fleet_placement_board.ships[1].segments[1].set_coordinate(Coordinate.new("b","2")).mark_as_hit
-    fleet_placement_board.ships[1].segments[2].set_coordinate(Coordinate.new("b","3")).mark_as_hit
-    fleet_placement_board.ships[1].segments[3].set_coordinate(Coordinate.new("b","4")).mark_as_hit
+  def possible_columns(ship, orientation)
+    orientation == "vertical" ? columns_data[0..9] : columns_data[0..(10-ship.length)]
+  end
 
-      # destroyer
-    fleet_placement_board.ships[2].segments[0].set_coordinate(Coordinate.new("c","1")).mark_as_hit
-    fleet_placement_board.ships[2].segments[1].set_coordinate(Coordinate.new("c","2")).mark_as_hit
-    fleet_placement_board.ships[2].segments[2].set_coordinate(Coordinate.new("c","3")).mark_as_hit
+  def rows_data
+    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+  end
 
-      # submarine
-    fleet_placement_board.ships[3].segments[0].set_coordinate(Coordinate.new("d","1")).mark_as_hit
-    fleet_placement_board.ships[3].segments[1].set_coordinate(Coordinate.new("d","2")).mark_as_hit
-    fleet_placement_board.ships[3].segments[2].set_coordinate(Coordinate.new("d","3")).mark_as_hit
+  def columns_data
+    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+  end
 
-      # patrol boat
-    fleet_placement_board.ships[4].segments[0].set_coordinate(Coordinate.new("e","1")).mark_as_hit
-    fleet_placement_board.ships[4].segments[1].set_coordinate(Coordinate.new("e","2"))
+  def check_if_valid_placement(starting_coordinate, ship, orientation)
+    resulting_coordinates = generate_test_coordinates(starting_coordinate, ship, orientation)
+    resulting_coordinates.each do | coordinate |
+      if coordinates_used.include?(coordinate)
+        return place_ship(ship)
+      end
+    end
+    @coordinates_used = coordinates_used.concat(resulting_coordinates)
+    assign_resulting_coordinates(ship, resulting_coordinates)
+  end
 
-    fleet_placement_board.update_data_with_ships
+  def generate_test_coordinates(starting_coordinate, ship, orientation)
+    test_coordinates = []
+    row = starting_coordinate[0]
+    column = starting_coordinate[1]
+    ship.length.times do 
+      test_coordinates.push([row, column])
+      orientation == "vertical" ? row += 1 : column += 1
+    end
+    test_coordinates
   end
 
+  def assign_resulting_coordinates(ship, resulting_coordinates)
+    ship.segments.each_with_index do | segment, index |
+      data_row = resulting_coordinates[index][0]
+      row_string = convert_row[data_row]
+      data_column = resulting_coordinates[index][1]
+      column_string = convert_column[data_column]
+      segment.set_coordinate(Coordinate.new(row_string, column_string))
+    end
+  end
+    
+  def convert_row
```

Comment:
These column and row methods represent an implicit coupling between your board and this class. One way to make it explicit is to ask the board for this information.